### PR TITLE
Feature/milestone 2.1

### PR DESCRIPTION
**Pull request by Lidya1234**
-Added a table for displaying books
-Added a form for adding books
-Created all the necessary files required for the project
-We made use of the redux-toolkit library in this project instead of vanilla redux because it is simpler, more powerful, and newer. Some file locations might change but the concept keeps being the same. Please take this into account when checking the code.

### CODE REVIEW

File: src/components/BooksList.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          </td>
+          <td>
+            {book.category}
+            {' '}
+          </td>
+        </tr>
+
+      ))}
+
+    </table>
```
Comment (uche-inyama) [inline src/components/BooksList.js]: In the booklist component, the instance of the book component rendered should receive 
book object as prop. Consequently, the component should look like so:

``` 
 <table>
      <thead>
        <tr>
          <th>ID</th>
          <th>Title</th>
          <th>Category</th>
        </tr>
      </thead>
      <tbody>
        {books.map((book) => (
          <Book key={book.id} book={book} />
        ))}
      </tbody>
    </table>
```

For the book id, you add it while creating your book in the book form component, using the [uuid package](https://www.npmjs.com/package/uuid)

File: src/components/Book.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const Book = (book = '') => {
+  <h1>{book}</h1>;
+};
+
+export default Book;
```
Comment (uche-inyama) [inline src/components/Book.js]: This is how the book component should look like.
```

const Book = ({ book: { id, title, category } }) => (
  <tr>
    <td>{id}</td>
    <td>{title}</td>
    <td>{category}</td>
  </tr>
);
```

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi team :wave: 

We need to restructure our files and make little modifications.

* [x] Booklist and bookform components should be moved to src/container directory.
* [x] There are two app.js files in your project. One is in the root directory and the other in the component folder; delete one of them.
* [x] Book component should be imported and rendered in Booklist component. 
* [x] In the booklist component, the instance of the book component rendered should receive 
 book object as prop.

That's it make these changes, then submit another form for a code review.

#### If there is a need for clarifications, kindly reach out on slack @ Inyama, Uchechukwu Henry.

### Happy coding :technologist: 

#### ``` Uchechukwu Henry Inyama ```

**Review by bellom — APPROVED** [review]
## Hi @Lidya1234  :clap: 

**Great work making all the necessary corrections from the previous review.**

# Status: Approved 👍 

## Congratulation :+1:  
_Go ahead and merge to dev and don't forget to submit the project completion form.
Good luck in the next milestone :slightly_smiling_face:_ 

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Pull request merged**